ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                    OGRERR_NONE)
+                {
+                    delete m_poSRS;
+                    m_poSRS = nullptr;
+                }
+            }
+
+            // If there is associated information
+            if (phMiraMonLayer->pMMBDXP)
+            {
+                if (!phMiraMonLayer->pMMBDXP->pfDataBase)
+                {
+                    if ((phMiraMonLayer->pMMBDXP->pfDataBase =
+                             fopen_function(phMiraMonLayer->pMMBDXP->szFileName,
+                                            "r")) == nullptr)
+                    {
+                        CPLDebug("MiraMon", "File '%s' cannot be opened.",
+                                 phMiraMonLayer->pMMBDXP->szFileName);
+                        bValidFile = false;
+                        return;
+                    }
+
+                    // First time we open the extended DBF we create an index
+                    // to fastly find all non geometrical features.
+                    phMiraMonLayer->pMultRecordIndex = MMCreateExtendedDBFIndex(
+                        phMiraMonLayer->pMMBDXP->pfDataBase,
+                        phMiraMonLayer->pMMBDXP->nRecords,
+                        phMiraMonLayer->pMMBDXP->FirstRecordOffset,
+                        phMiraMonLayer->pMMBDXP->BytesPerRecord,
+                        phMiraMonLayer->pMMBDXP
+                            ->pField[phMiraMonLayer->pMMBDXP->IdGraficField]
+                            .AcumulatedBytes,
+                        phMiraMonLayer->pMMBDXP
+                            ->pField[phMiraMonLayer->pMMBDXP->IdGraficField]
+                            .BytesPerField,
+                        &phMiraMonLayer->isListField, &phMiraMonLayer->nMaxN);
+
+                    // Creation of maximum number needed for processing
+                    // multiple records
+                    if (phMiraMonLayer->pMultRecordIndex)
+                    {
+                        papszValues = static_cast<char **>(
+                            CPLCalloc((size_t)phMiraMonLayer->nMaxN + 1,
+                                      sizeof(*papszValues)));
+
+                        padfValues = static_cast<double *>(
+                            CPLCalloc((size_t)phMiraMonLayer->nMaxN,
+                                      sizeof(*padfValues)));
+                    }
+
+                    phMiraMonLayer->iMultiRecord =
+                        -2;  // No option iMultiRecord
+                    const char *szMultiRecord =
+                        CSLFetchNameValue(papszOpenOptions, "iMultiRecord");
+                    if (phMiraMonLayer->isListField && szMultiRecord)
+                    {
+                        if (!strcasecmp(szMultiRecord, "Last"))
+                            phMiraMonLayer->iMultiRecord = -1;
+                        else if (!strcasecmp(szMultiRecord, "JSON"))
+                            phMiraMonLayer->iMultiRecord = -3;
```

Comment:
it would be best to use a symbolic name like MM_MULTIRECORD_LAST and MM_MULTIRECORD_JSON or something like that, here, and where they are later used